crates/ruff_cli/src/printer.rs

```diff
@@ -118,19 +119,14 @@ impl Printer {
                     writeln!(writer, "Found {remaining} error{s}.")?;
                 }
 
-                if show_fix_status(self.autofix_level) {
-                    let num_fixable = diagnostics
-                        .messages
-                        .iter()
-                        .filter(|message| message.fix.is_some())
-                        .count();
-                    if num_fixable > 0 {
-                        writeln!(
-                            writer,
-                            "[{}] {num_fixable} potentially fixable with the --fix option.",
-                            "*".cyan(),
-                        )?;
-                    }
+                if let Some((safe_remaining, unsafe_remaining)) =
+                    applicable_fixes(self.autofix_level, diagnostics)
```

Comment:
I think it would be good to compute the diagnostic counts once instead of once in `write_summary_text` and once in `write_once` to avoid iterating the diagnostics twice (could be good to run the cpython benchmark before/after your change)